OpenEdXMobile/src/main/java/org/edx/mobile/course/CourseAPI.kt

```diff
@@ -0,0 +1,324 @@
+import org.edx.mobile.model.course.CourseDates
+import org.edx.mobile.model.course.CourseDetail
+import org.edx.mobile.model.course.CourseStatus
+import org.edx.mobile.model.course.CourseStructureV1Model
+import org.edx.mobile.model.course.DiscussionBlockModel
+import org.edx.mobile.model.course.DiscussionData
+import org.edx.mobile.model.course.EnrollBody
+import org.edx.mobile.model.course.HtmlBlockModel
+import org.edx.mobile.model.course.IBlock
+import org.edx.mobile.model.course.ResetCourseDates
+import org.edx.mobile.model.course.VideoBlockModel
+import org.edx.mobile.model.course.VideoData
+import org.edx.mobile.module.prefs.UserPrefs
+import org.edx.mobile.util.Config
+import org.edx.mobile.view.common.TaskProgressCallback
+import org.greenrobot.eventbus.EventBus
+import org.json.JSONObject
+import retrofit2.Call
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CourseAPI @Inject constructor(
+    private var config: Config,
+    private val courseService: CourseService,
+    private val userPrefs: UserPrefs
+) {
+    /**
+     * @return Enrolled courses of given user.
+     */
+    val enrolledCourses: Call<EnrollmentResponse>
+        get() = courseService.getEnrolledCourses(
+            null,
+            userPrefs.profile.username,
+            config.apiUrlVersionConfig.enrollmentsApiVersion,
+            config.organizationCode
+        )
+
+    /**
+     * @return Enrolled courses of given user without stale response.
+     */
+    val enrolledCoursesWithoutStale: Call<EnrollmentResponse>
+        get() = courseService.getEnrolledCourses(
+            "stale-if-error=0",
+            userPrefs.profile.username,
+            config.apiUrlVersionConfig.enrollmentsApiVersion,
+            config.organizationCode
+        )
+
+    /**
+     * @return Enrolled courses of given user, only from the cache.
+     */
+    val enrolledCoursesFromCache: Call<EnrollmentResponse>
+        get() = courseService.getEnrolledCoursesFromCache(
+            userPrefs.profile.username,
+            config.apiUrlVersionConfig.enrollmentsApiVersion,
+            config.organizationCode
+        )
+
+    fun enrollInACourse(courseId: String, emailOptIn: Boolean): Call<ResponseBody> {
+        return courseService.enrollInACourse(EnrollBody(courseId, emailOptIn))
+    }
+
+    fun getCourseList(page: Int): Call<Page<CourseDetail>> {
+        val username = if (userPrefs.loginPrefs.iaUserLoggedIn())
+            userPrefs.profile.username else null
+        return courseService.getCourseList(username, true, config.organizationCode, page)
+    }
+
+    fun getCourseDetail(courseId: String): Call<CourseDetail> {
+        // Empty courseId will return a 200 for a list of course details, instead of a single course
+        require(courseId.isNotEmpty())
+        return courseService.getCourseDetail(courseId, userPrefs.profile.username)
+    }
+
+    fun getCourseStructure(
+        blocksApiVersion: String,
```

Comment:
IMO, we can use the local inject Config variable to get the required version of API instead.
i.e, `apiVersion.blocksApiVersion`